tests/aws/services/kms/test_kms.py

```diff
@@ -1465,6 +1465,159 @@ def test_rotate_key_on_demand_raises_error_given_key_with_imported_key_material(
             aws_client.kms.rotate_key_on_demand(KeyId=key_id)
         snapshot.match("error-response", e.value.response)
 
+    @markers.aws.validated
+    def test_list_key_rotations_after_on_demand_rotation(
+        self, kms_create_key, aws_client, snapshot
+    ):
+        key_id = kms_create_key(KeyUsage="ENCRYPT_DECRYPT", KeySpec="SYMMETRIC_DEFAULT")["KeyId"]
+
+        snapshot.add_transformer(
+            snapshot.transform.key_value("KeyMaterialId", reference_replacement=False)
+        )
+
+        response_initial_rotation = aws_client.kms.list_key_rotations(KeyId=key_id)
+        snapshot.match("initial-rotations", response_initial_rotation)
+
+        aws_client.kms.rotate_key_on_demand(KeyId=key_id)
+
+        def _rotation_completed():
+            status = aws_client.kms.get_key_rotation_status(KeyId=key_id)
+            return "OnDemandRotationStartDate" not in status
+
+        assert poll_condition(condition=_rotation_completed, timeout=10, interval=1)
+
+        response_after_on_demand_rotation = aws_client.kms.list_key_rotations(KeyId=key_id)
+        snapshot.match("after-on-demand-rotation", response_after_on_demand_rotation)
+
+        assert (
+            len(response_after_on_demand_rotation["Rotations"])
+            == len(response_initial_rotation["Rotations"]) + 1
+        )
+
+    @markers.aws.validated
+    def test_list_key_rotations_different_key_states(self, kms_create_key, aws_client, snapshot):
+        key_id = kms_create_key(KeyUsage="ENCRYPT_DECRYPT", KeySpec="SYMMETRIC_DEFAULT")["KeyId"]
```

Comment:
Since `schedule_key_deletion` is used explicitly in this test, you should rather not use a fixture here since it also does a cleanup later and use `aws_client.kms.create_key`